specs/agents/field-limits.md

```diff
@@ -0,0 +1,30 @@
+## Field limits
+
+The maximum length of metadata, transaction, span, et al fields are determined
+by the [APM Server Events Intake API schema](https://www.elastic.co/guide/en/apm/server/current/events-api.html).
+Fields that are names or identifiers of some resource of typically limited to
+1024 unicode characters.
+
+### `long_field_max_length` configuration
+
+Some APM event fields are not limited in the APM server intake API schema.
+Agents SHOULD limit the maximum length of the following fields by truncating.
+
+- `{transaction,error}.context.request.body`
+- `{transaction,span,error}.context.message.body`
+- `span.context.db.statement`
+- `error.exception.message`
+- `error.log.message`
+
+Agents MAY support the `long_field_max_length` configuration option to allow
+the user to configure this maximum length.
+
+|                |   |
+|----------------|---|
+| Type           | [`Size`](./configuration.md#configuration-value-types) |
```

Comment:
Is there a particular reason to limit by bytes vs number of characters? I find the latter to be more transparent for users to understand.
The actual size in ES storage depends on a lot of factors, such as compression, index settings and uniqueness of the value.

Reply:
I agree on characters being clearer. Partially, I had been lazily following what the Node.js Agent's current `errorMessageMaxLength` does -- which is bytes.

Does the Java agent (or any other agent you are aware of) have a config type that takes size unit *prefixes*, but not the 'b' signifying a number of "bytes"? The other reason I'd reached for the existing "Size" type here was to allow users to use the "k", "m" prefixes: `long_field_max_length: '20kb'` rather than `long_field_max_length: 20000`. However with the 'b' for bytes, this is definitely misleading.

Defining a new "Count" or "Length" type -- `^(\d+)(|k|m)` -- at https://github.com/elastic/apm/blob/master/specs/agents/configuration.md is probably overkill for this. Let me know if you disagree. Otherwise I'll set the type to `Integer`.

Reply:
> Otherwise I'll set the type to `Integer`.

Done